### ISSUE TITLE
bug([mat-dialog]): screenreader navigation order not logical

### ISSUE THREAD

**Issue by soshochang** [angular/components#19338]
*This is similar to this issue that I filed earlier https://github.com/angular/components/issues/19253

#### Reproduction

Steps to reproduce:
*must navigate with VO commands to reproduce, not keyboard tab

turn on mac voiceover (system preferences -> accessibility -> voiceover --> enable voiceover)
at url above use voiceover to navigate to the mat-dialog button

select the mat-dialog button
navigate to next element using voiceover (not tab)

*Voiceover command == VO == caps lock key or control + option by default
VO + shift + down arrow == enter the element to navigate children of element
VO + space == select the element
VO + left/right arrow == navigate to prev/next element

#### Expected Behavior

What behavior were you expecting to see?
-screen reader focus should be on the mat-dialog

#### Actual Behavior

What behavior did you actually see?
-the screen reader focus is on the entire page content, if you navigate forward one more item, then it is on the mat-dialog
-also you dismiss the mat-dialog by hitting a cancel button in the mat-dialog, the screen reader focus returns to the beginning of the page rather than going to back to the mat-dialog button that launched the dialog

#### Environment

- Angular: 9.10
- CDK/Material: 9.22
- Browser(s): chrome
- Operating System (e.g. Windows, macOS, Ubuntu): macOS

**Comment by jelbourn** [comment]
Similar to #19253, this only happens on Chrome, not Safari. I tested VoiceOver with the internal Google ARIA blueprint for dialog, and VoiceOver + Chrome still behaves the same way there. This leads me to believe this is a VoiceOver + Chrome issue that we can't do anything to fix at the library level. 

@soshochang are there any specific changes you propose to address this at the library level?

**Comment by wagnermaciel** [comment]
Closing because this issue has been inactive for 2 weeks or more after follow up was requested.
